app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx

```diff
@@ -37,106 +36,102 @@ interface PropsWithoutHref {
 /**
  * Base props for a button.
  */
-export interface BaseButtonProps {
+export interface BaseButtonProps extends Omit<twv.VariantProps<typeof BUTTON_STYLES>, 'iconOnly'> {
   /** Falls back to `aria-label`. Pass `false` to explicitly disable the tooltip. */
   readonly tooltip?: React.ReactNode
-  readonly loading?: boolean
-  /**
-   * The variant of the button
-   */
-  readonly variant: Variant
   /**
    * The icon to display in the button
    */
-  readonly icon?: string
-  /**
-   * The size of the button
-   */
-  readonly size?: Size
-  readonly rounding?: Roundings
+  readonly icon?: string | null
   /**
-   * Button takes the full width of its container
+   * When `true`, icon will be shown only when hovered.
    */
-  readonly fullWidth?: boolean
+  readonly showIconOnHover?: boolean
```

Comment:
do we actually want this? this sounds like it will cause the size of the element to change

Reply:
Afair we use opacity to hide the icon. Ideally, I guess we might want to provide an external `twv variants`  scheme that satisfies the button's one(e.g. that extends the button's), but I still have no clue how to do that properly. By doing so, we can customize any slot and no longer need to use these properties (it's less safe though)